launchdarkly/resource_feature_flag.go

```diff
@@ -0,0 +1,250 @@
+package launchdarkly
+
+import (
+	"github.com/hashicorp/terraform/helper/schema"
+)
+
+func resourceFeatureFlag() *schema.Resource {
+	return &schema.Resource{
+		Create: resourceFeatureFlagCreate,
+		Read:   resourceFeatureFlagRead,
+		Update: resourceFeatureFlagUpdate,
+		Delete: resourceFeatureFlagDelete,
+
+		Schema: map[string]*schema.Schema{
+			"project_key": &schema.Schema{
+				Type:     schema.TypeString,
+				Required: true,
+			},
+			"name": &schema.Schema{
+				Type:     schema.TypeString,
+				Required: true,
+			},
+			"key": &schema.Schema{
+				Type:     schema.TypeString,
+				Required: true,
+			},
+			"description": &schema.Schema{
+				Type:     schema.TypeString,
+				Optional: true,
+			},
+			"temporary": &schema.Schema{
+				Type:     schema.TypeBool,
+				Optional: true,
+				Default:  true,
+			},
+			"include_in_snippet": &schema.Schema{
+				Type:     schema.TypeBool,
+				Optional: true,
+				Default:  false,
+			},
+			"tags": &schema.Schema{
+				Type:     schema.TypeList,
+				Optional: true,
+				Elem:     &schema.Schema{Type: schema.TypeString},
+			},
+			"custom_properties": &schema.Schema{
+				Type:     schema.TypeList,
+				Optional: true,
+				Elem: &schema.Resource{
+					Schema: map[string]*schema.Schema{
+						"key": {
+							Type:     schema.TypeString,
+							Required: true,
+						},
+						"name": &schema.Schema{
+							Type:     schema.TypeString,
+							Required: true,
+						},
+						"value": &schema.Schema{
+							Type:     schema.TypeList,
+							Required: true,
+							Elem:     &schema.Schema{Type: schema.TypeString},
+						},
+					},
+				},
+			},
+		},
+	}
+}
+
+func resourceFeatureFlagCreate(d *schema.ResourceData, m interface{}) error {
+	client := m.(Client)
+
+	project := d.Get("project_key").(string)
```

Comment:
Any of these will panic if the data is not of the correct type, is this the expected behavior?

Reply:
This is usually how it's done in terraform providers. Terraform attributes are expected to be of the correct type or panic.

Reply:
I'm curious as to why, but it's okay I guess if you are used to that.

Reply:
We define the schema for Terraform, so I think it's ok to expect it to pass proper values afterwards. Otherwise that already verbosy-ish code would be even more verbose 😁